spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDelete.scala

```diff
@@ -77,33 +101,30 @@ object RewriteDelete extends Rule[LogicalPlan] with PredicateHelper with Logging
     PushDownUtils.pushFilters(scanBuilder, normalizedPredicates)
 
     val scan = scanBuilder.build()
-    val scanRelation = DataSourceV2ScanRelation(table, scan, output)
+    val scanRelation = DataSourceV2ScanRelation(table, scan, toOutputAttrs(scan.readSchema(), output))
```

Comment:
Spark's contract is that the scan's schema is the one that should be used, not the original table schema. This allows the merge scan to return the extra `_file` and `_pos` columns and matches the behavior of normal scans that are configured with `PushDownUtils.pruneColumns`.